server/portal/apps/projects/management/commands/projects_id.py

```diff
@@ -1,96 +1,72 @@
                     ''
                 )
                 if '-' not in prj_id:
                     continue
                 _, prj_id = prj_id.rsplit('-')
-                try:
-                    prj_id = int(prj_id)
-                except ValueError:
-                    # prj_id is not an int
-                    pass
+                prj_id = int(prj_id)
 
-                if prj_id > latest:
+                if prj_id > latest and (ignore_above_value is None or prj_id < ignore_above_value):
                     latest = prj_id
-            if prj_id == -1:
-                done = True
-            else:
-                offset += 100
-                prjs = Project.listing(
-                    service_account(),
-                    offset=offset,
-                    limit=limit
-                )
+
+            offset += limit
+            if len(prjs) < limit:
+                break
         return latest
 
-    def _get_latest_project_directory(self):  # pylint:disable=no-self-use
-        """Get latest agave project directory."""
-        offset = 0
-        limit = 100
-        _file = BaseFile(
-            service_account(),
-            system=settings.PORTAL_PROJECTS_ROOT_SYSTEM_NAME,
-            path='/'
-        )
-        dirs = _file.children(offset=offset, limit=limit)
-        done = False
+    @staticmethod
+    def get_latest_project_directory(ignore_above_value=None):
+        """Get latest agave project directory.
+
+        :param ignore_above_value: If provided, then ignore projects ids that are greater than or equal to this value.
+        """
         latest = -1
-        while not done:
-            dir_id = -1
-            for _dir in dirs:
-                if ('-' not in _dir.name or
-                        not _dir.name.startswith(settings.PORTAL_PROJECTS_ID_PREFIX)):
-                    continue
-                _, dir_id = _dir.name.rsplit('-', 1)
-                try:
-                    dir_id = int(dir_id)
-                except ValueError:
-                    # dir_id is not an int
-                    pass
-
-                if dir_id > latest:
-                    latest = dir_id
-
-            if dir_id == -1:
-                done = True
-            else:
-                offset += 100
-                _file.set_children(None)
-                dirs = _file.children(offset=offset, limit=limit)
+        for f in iterate_listing(service_account(),
+                                 system=settings.PORTAL_PROJECTS_ROOT_SYSTEM_NAME,
+                                 path='/'):
+            name = f["name"]
+            if '-' not in name or not name.startswith(settings.PORTAL_PROJECTS_ID_PREFIX):
+                continue
+            _, dir_id = name.rsplit('-', 1)
+            dir_id = int(dir_id)
+            if dir_id > latest and (ignore_above_value is None or dir_id < ignore_above_value):
+                latest = dir_id
         return latest
 
     def add_arguments(self, parser):
```

Comment:
Should we include arguments for:
- `--update-using-project-id`
- `--update-using-max-value`, where it looks for the max safe value between storage systems, project ids, and directories?

Reply:
i like the idea of `--update-using-max-value`; that is what we really want (a way to get a "safe" value).

Instead of supporting 4 types of update (`--update-using-storage-system-id`, `--update-using-project-id`, `--update`, `--update-using-max-value`), could we just go with these two:

- `--update-using-max-value`
-  `--update`

--------
if so, any other name ideas for `--update-using-max-value`. i've just stared at the screen for a few minutes and can't think of anything so maybe that is good enough. 🤔   help could be _"Update project id DB value to use the largest storage system project id or largest project folder id (i.e. the maximum value of those two numbers)"_

Reply:
Great idea. And would be good to keep the ability to ignore values over a certain threshold